src/hipscat/pixel_math/partition_stats.py

```diff
@@ -0,0 +1,150 @@
+"""Utilities for generating and manipulating object count histograms"""
+
+import healpy as hp
+import numpy as np
+import pandas as pd
+
+
+def empty_histogram(highest_order):
+    """Use numpy to create an histogram array with the right shape, filled with zeros.
+    Args:
+        highest_order (int): the highest healpix order (e.g. 0-10)
+    Returns:
+        one-dimensional numpy array of long integers, where the length is equal to
+        the number of pixels in a healpix map of target order, and all values are set to 0.
+    """
+    return np.zeros(hp.order2npix(highest_order), dtype=np.ulonglong)
+
+
+def generate_histogram(
+    data: pd.DataFrame,
+    highest_order,
+    ra_column="ra",
```

Comment:
Consider making optional kwargs kwarg only, it will make the API much nicer and also much easier to extend and maintain. This is a comment for here and everywhere else, too.

Reply:
This is tough for me, and I'm still unpacking my reaction.

I think I'm making this optional because I want to provide some reasonable default, but really, I think the better thing to do is bundle the values and defaults in the CatalogParameters class and pass those around. I'll make that change in a follow-up refactor.

Reply:
Oh, I didn't suggest to make them mandatory, but to make them kwarg only (https://peps.python.org/pep-3102/), so if you need to add a new kwarg in the future, you can add anywhere without breaking API. Also, I think it's good practice to require it from users to spell out things, at least the optional parameters especially when there are more than a few params altogether as that would make it somewhat more bug proof:

```def generate_histogram(data: pd.DataFrame, highest_order, *, ra_column="ra", dec_column="dec"):```